### PR TITLE
Delete user account

### PR DESCRIPTION
**Pull request by Nash0x7E2**
This PR adds the ability for users to remove their account. 
- Migrated delete account from cognito auth to user service 
- Updated references 
- Fixed minor lint rules and code clean up

### CODE REVIEW

File: lib/backend/repositories/user_repo.dart [hunk 1]
```diff
@@ -229,6 +230,17 @@ class UserRepo {
     return false;
   }
 
+  Future<void> deleteUserAccount(String userAddress, String password) {
+    if (userAddress.isNotEmpty &&
+        userAddress != null &&
```
Comment (orestesgaolin) [inline lib/backend/repositories/user_repo.dart:235]: if it's not empty then it's not null (the order of comparison matters)

File: lib/backend/services/user_service.dart [hunk 2]
```diff
@@ -518,4 +518,19 @@ class UserServiceCentralized implements UserService {
     final UserData _userData = UserData.fromJson(_responseMap);
     return _userData;
   }
+
+  Future<void> deleteUser(String userAddress, String password) async {
+    final Map<String, dynamic> _body = <String, String>{
+      'password': password,
+    };
```
Comment (orestesgaolin) [inline lib/backend/services/user_service.dart:525]: password is not required (cc: @jdeepee was supposed to fix this in docs)

File: lib/backend/repositories/user_repo.dart (hunk 1 above)
Comment (Nash0x7E2) [reply lib/backend/repositories/user_repo.dart:235]: Right, good catch.

File: lib/backend/services/user_service.dart (hunk 2 above)
Comment (Nash0x7E2) [reply lib/backend/services/user_service.dart:525]: Cool, will remove

**Comment by Nash0x7E2** [comment]
Yes, will confirm with @jdeepee over slack.

**Comment by Nash0x7E2** [comment]
@orestesgaolin Changes made 👍

**Pull request merged**